### DISCUSSION TITLE
How to provide a custom python DLL file?

### DISCUSSION THREAD

**Discussion by Kreonid** [pyinstaller/pyinstaller#6810] [Help]
I would like to provide my own python DLL, with a different name than usual. For example, I want my custom DLL named super_python.dll to be picked by PyInstaller. What's the best way to do this?

Currently I see in the logs that Pyinstaller finds the DLL automatically:
`INFO: Using Python library c:\users\me\appdata\local\programs\python\python37\python37.dll`

**Comment by bwoodsend** [comment]
In all honesty, the *best* way would be not to do it in the first place. Write an extension module if you want to extend Python.

If you insist on doing this and you're using a spec file then you could inject something like the code below immediately after the `a = Analysis(...)` bit:
```python
for (i, (dest, source, type_)) in enumerate(a.binaries):
     if dest == "python37.dll":
         a.binaries[i] = (dest, "/full/path/to/super_python.dll", type_)
```
